### PR TITLE
Allow passing headers to proxy

### PR DESCRIPTION
**Pull request by esvinson**
There was not way to pass any headers to the proxy CONNECT request for the TunnelProxy.  Because of this there was no way to authenticate with a proxy that required basic authentication.

I bumped up the alpine version so it'd have a more recent version of tinyproxy that included the BasicAuth support.

I added a secondary docker container for testing authenticated proxy requests.

I added tests for authenticated proxy requests for both the Unsafe and Tunnel proxies.

### CODE REVIEW

**Comment by ericmj** [comment]
I think it would be better to add a `:proxy_headers` option instead of adding the headers to the `opts` element in `:proxy`. `:proxy` is about the connection so the headers do not really fit imo. WDYT?

**Comment by whatyouhide** [comment]
Yes, let's do what @ericmj suggested please 🙃

**Comment by esvinson** [comment]
> I think it would be better to add a `:proxy_headers` option instead of adding the headers to the `opts` element in `:proxy`. `:proxy` is about the connection so the headers do not really fit imo. WDYT?

In reference to basic auth with a tunnel based proxy, the headers need to be there in the `CONNECT` request.  So the proxy headers would be part of the connection.

Are you suggesting just adding an option to support `%{proxy: {scheme, hostname, port, opts, proxy_headers}}`, or were you suggesting passing the `:proxy_headers` as part of the destination request?  The later won't work for tunnel based proxies as the header needs to be part of the `CONNECT` request to the proxy and not part of the request to the destination.

**Comment by esvinson** [comment]
Updated to use `:proxy_headers` as part of the opts for `Mint.HTTP.connect/4`, working with Unsafe and Tunnel proxies.
```
Mint.HTTP.connect(:https, "httpbin.org", 443,
  proxy: {:http, "localhost", 8889, []},
  proxy_headers: [{"proxy-authorization", "basic #{auth64}"}]
)
```

**Comment by ericmj** [comment]
Looks great, we are just missing docs then we are good to go.

File: lib/mint/unsafe_proxy.ex [hunk 1]
```diff
@@ -78,6 +80,7 @@ defmodule Mint.UnsafeProxy do
         body \\ nil
       ) do
     path = request_line(conn, path)
+    headers = Keyword.merge(headers, Map.get(conn, :proxy_headers, []))
```
Comment (whatyouhide) [inline lib/mint/unsafe_proxy.ex]: `conn` always has the `:proxy_headers` field here, no need to default to `[]`. I'll change it.

**Comment by whatyouhide** [comment]
@ericmj I will work on docs. :)

**Pull request merged**